web/api/v1/api.go

```diff
@@ -496,6 +496,23 @@ func (api *API) remoteRead(w http.ResponseWriter, r *http.Request) {
 			OldestInclusive: from,
 			NewestInclusive: through,
 		}))
+		externalLabels := api.config().GlobalConfig.ExternalLabels.Clone()
+		for _, ts := range resp.Results[i].Timeseries {
+			globalUsed := map[string]struct{}{}
+			for _, l := range ts.Labels {
+				if _, ok := externalLabels[model.LabelName(l.Name)]; ok {
+					globalUsed[l.Name] = struct{}{}
+				}
+			}
+			for ln, lv := range externalLabels {
+				if _, ok := globalUsed[string(ln)]; !ok {
+					ts.Labels = append(ts.Labels, &remote.LabelPair{
+						Name:  string(ln),
+						Value: string(lv),
+					})
+				}
+			}
+		}
```

Comment:
Would it be easier to add them to the temporary matrix that is returned from `remote.IteratorsToMatrix`?  Then we wouldn't need the extra map etc.

Reply:
As I mentioned in person, I copied this approach from the federation code and I'm not sure it's worth worrying too much about this for 1.8, as the 2.0 implementation will be different.

Reply:
Okay, LGTM then.